pilot/cmd/pilot-agent/main.go

```diff
@@ -489,3 +500,37 @@ func main() {
 		os.Exit(-1)
 	}
 }
+
+// GetClusterDomainName returns cluster domain name. For efficiency, it runs just once
+// all subsequent calls will just return domain name value.
+func GetClusterDomainName() string {
```

Comment:
This should not be a public method, and both should be declared up above init with all the other private functions.

Reply:
well, personally I think main.go is not the right place for this func, it should be in some `util` package and called from there. I placed them here just because 1. if you guys going to accept this solution and 2. I was not sure which common utility or tools package I could use.
Please let me know if you would be ok if I move them into some common utility package.